plutus-core/untyped-plutus-core/bench/Main.hs

```diff
@@ -0,0 +1,190 @@
+                           ]
+    , bgroup "create/250" [ bench "bral" $ whnf (extendB B.nil) 250
+                           , bench "ral" $ whnf (extendR R.empty) 250
+                           , bench "imap" $ whnf (extendI I.empty) 250
+                            ]
+
+
+    , bgroup "query/front/100" [ bench "bral" $ whnf (queryFrontB 100) b100
+                                , bench "ral" $ whnf (queryFrontR 100) r100
+                                , bench "imap" $ whnf (queryFrontI 100) i100
+                                ]
+    , bgroup "query/front/250" [ bench "bral" $ whnf (queryFrontB 250) b250
+                                , bench "ral" $ whnf (queryFrontR 250) r250
+                                , bench "imap" $ whnf (queryFrontI 250) i250
+                                ]
+
+    , bgroup "query/back/100" [ bench "bral" $ whnf (queryBackB 100) b100
+                               , bench "ral" $ whnf (queryBackR 100) r100
+                               , bench "imap" $ whnf (queryBackI 100) i100
+                               ]
+    , bgroup "query/back/250" [ bench "bral" $ whnf (queryBackB 250) b250
+                               , bench "ral" $ whnf (queryBackR 250) r250
+                               , bench "imap" $ whnf (queryBackI 250) i250
+                               ]
+
+
+    , bgroup "query/rand/100" [ bench "bral" $ whnf (uncurry queryRandB) (rand100, b100)
+                               , bench "ral" $ whnf (uncurry queryRandR) (rand100, r100)
+                               , bench "imap" $ whnf (uncurry queryRandI) (rand100, i100)
+                               ]
+    , bgroup "query/rand/250" [ bench "bral" $ whnf (uncurry queryRandB) (rand250, b250)
+                               , bench "ral" $ whnf (uncurry queryRandR) (rand250, r250)
+                               , bench "imap" $ whnf (uncurry queryRandI) (rand250, i250)
+                               ]
+
+    , bgroup "create100/front100/cons100/back100/cons100/rand100"
+            [ bench "bral" $ whnf (uncurry $ mixB 100 100 100 100) (rand100, b100)
+            , bench "ral" $ whnf (uncurry $ mixR 100 100 100 100) (rand100, r100)
+            , bench "imap" $ whnf (uncurry $ mixI 100 100 100 100) (rand100, i100)
+            ]
+    , bgroup "create250/front100/cons100/back100/cons100/rand250"
+            [ bench "bral" $ whnf (uncurry $ mixB 100 100 100 100) (rand250, b250)
+            , bench "ral" $ whnf (uncurry $ mixR 100 100 100 100) (rand250, r250)
+            , bench "imap" $ whnf (uncurry $ mixI 100 100 100 100) (rand250, i250)
+            ]
+
+
+    ]
+  where
+        b100 = extendB B.nil 100
+        r100 = extendR R.empty 100
+        i100 = extendI I.empty 100
+        b250 = extendB B.nil 250
+        r250 = extendR R.empty 250
+        i250 = extendI I.empty 250
+        rand100 = take 100 $ randomRs (0,100-1) g
+        rand250 = take 250 $ randomRs (0,250-1) g
+        g = mkStdGen 59950
```

Comment:
Do we want to fix the seed here? I guess we do to make the benchmarks deterministic. Worth a comment.